### PR TITLE
fix: allow to update rpc urls

### PR DESCRIPTION
**Pull request by alaibe**
* in all cases update the original urls in case they changed with a new api key
* Update current urls only if the current one is the same as the previous original one and the original one did change

### CODE REVIEW

**Review by saledjenic — APPROVED** [review]

File: rpc/network/network.go [hunk 1]
```diff
@@ -148,7 +143,22 @@ func (nm *Manager) Init(networks []params.Network) error {
 						errors += fmt.Sprintf("error updating network fallback_url for ChainID: %d, %s", currentNetworks[j].ChainID, err.Error())
 					}
 				}
-				err := nm.UpdateRPCURLs(currentNetworks[j].ChainID, networks[i].RPCURL, networks[i].OriginalRPCURL, networks[i].FallbackURL, networks[i].OriginalFallbackURL)
+
+				if networks[i].OriginalRPCURL != currentNetworks[j].OriginalRPCURL && currentNetworks[j].RPCURL == currentNetworks[j].OriginalRPCURL {
+					err := nm.UpdateRPCURL(networks[i].ChainID, networks[i].OriginalRPCURL)
```
Comment (saledjenic) [inline rpc/network/network.go]: @alaibe wondering if it makes sense to do these queries as transactions instead of executing each query separately? 

Cause in the current implementation if an error occurs at some point, we return the errors list but do not revert the changes.

File: rpc/network/network.go [hunk 2]
```diff
@@ -193,9 +203,18 @@ func (nm *Manager) UpdateRelatedChainID(chainID uint64, relatedChainID uint64) e
 	return err
 }
 
-func (nm *Manager) UpdateRPCURLs(chainID uint64, rpcURL, originalRPCURL, fallbackURL, originalFallbackURL string) error {
-	_, err := nm.db.Exec(`UPDATE networks SET rpc_url = ?, original_rpc_url = ?, fallback_url = ?, original_fallback_url = ?  WHERE chain_id = ?`,
-		rpcURL, originalRPCURL, fallbackURL, originalFallbackURL, chainID)
+func (nm *Manager) UpdateRPCURL(chainID uint64, rpcURL string) error {
```
Comment (saledjenic) [inline rpc/network/network.go]: @alaibe guess that we don't need to make this and the calls below public (they all should start with a small first letter)?

Comment (alaibe) [reply rpc/network/network.go]: Done

File: rpc/network/network.go (hunk 1 above)
Comment (alaibe) [reply rpc/network/network.go]: yes currently we only log the errors, are they really independent query or are they a group of query? good question. I am wondering if we should not return the error

Comment (saledjenic) [reply rpc/network/network.go]: They are independent queries that are executed successively and theoretically if all of them as an entirety introduce the change then if an error occurs among any two we should roll back. That's why I proposed to make a transaction and get back as soon as we face an error.

But in practice, I don't have anything against this approach you did, since the only error which may happen in that loop is the DB error, and in reality if that happens we don't have so much to do, cause the app won't run anymore.

So you can leave it as it's now, it's already approved. :)
